linera-execution/src/runtime.rs

```diff
@@ -745,19 +891,22 @@ impl ContractSyncRuntime {
             signer,
         });
         let runtime = ContractSyncRuntime::new(runtime);
-        let execution_outcome = {
+        let execution_result = {
             let mut code = code.instantiate(runtime.clone())?;
             match action {
-                UserAction::Initialize(context, argument) => code.initialize(context, argument)?,
+                UserAction::Initialize(context, argument) => code.initialize(context, argument),
                 UserAction::Operation(context, operation) => {
-                    code.execute_operation(context, operation)?
+                    code.execute_operation(context, operation)
                 }
-                UserAction::Message(context, message) => code.execute_message(context, message)?,
+                UserAction::Message(context, message) => code.execute_message(context, message),
             }
         };
+        // Ensure the `loaded_applications` are cleared to prevent circular references in the
+        // `runtime`
         let mut runtime = runtime
             .into_inner()
             .expect("Runtime clones should have been freed by now");
+        let execution_outcome = execution_result?;
```

Comment:
This should matter any more because all errors are final, but sure.

Reply:
When I tested it would hang without this line on tests which returned an error.

Reply:
Oh wow, ok!